common/dynamicconfig/source.go

```diff
@@ -0,0 +1,99 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package dynamicconfig
+
+import (
+	enumspb "go.temporal.io/api/enums/v1"
+)
+
+type (
+	// Source is a source of dynamic configuration. The default Source, fileSource, reads from
+	// a file in the filesystem, and refreshes it periodically. You can extend the server with
+	// an alternate Source using ServerOptions.
+	Source interface {
```

Comment:
Can we keep using Client as name for this thing? It is perfectly fine that the only job a client does is fetch value for a given key from external source.

Reply:
Ehh, I disagree about the name: the most common meaning of "client" is "a thing that talks to a server", which this could be but usually isn't, so it's not that. The next most common meaning is something like "thing that calls an API", which also doesn't fit. With that meaning, a "dynamic config client" is a user of the dynamic config interface, not an implementation of it.

Also, is there any compatibility advantage to keeping the name? The contents of the interface are totally different

Reply:
All good, except that the config will continue to use dyanmicConfigClient, but I'm ok with it.

Reply:
we can also support both config keys if you think it's valuable

Reply:
not worth it, will cause more confusion.